src/raftstore/store/peer.rs

```diff
@@ -966,15 +965,25 @@ impl Peer {
             let region = snap_data.take_region();
             // For merge process, when applying snapshot or create new peer the stale source peer is destroyed asynchronously.
             // So here checks whether there is any overlap, if so, wait and do not handle raft ready.
-            if let Some(r) = region_ranges
-                .range((Excluded(enc_start_key(&region)), Unbounded::<Vec<u8>>))
+            // Note that: below logic can not find the overlap region with range (a, b] that a < snap_region.end_key < b,
+            // but it is okay cause this case would never happen.
```

Comment:
Better add some `assert`s to ensure that even it happens due to bugs, data will not get corrupted.